.github/workflows/maven.yml

```diff
@@ -0,0 +1,211 @@
+name: Java CI with Maven and native builds
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  # Java version to use for the release
+  RELEASE_JAVA_VERSION: 11
+  UBUNTU_DIST: focal
+  UBUNTU_MIRROR_AMD64: http://azure.archive.ubuntu.com/ubuntu
+  DOCKER_BUILDKIT: 1
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  javah:
+
+    runs-on: ubuntu-latest
+
+    name: Compile and generate JNI headers
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Set up JDK ${{ env.RELEASE_JAVA_VERSION }}
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: ${{ env.RELEASE_JAVA_VERSION }}
+          cache: maven
+
+      - name: Build with Maven
+        run: mvn -B compile
+
+      - name: Upload JNI headers
+        uses: actions/upload-artifact@v4
+        with:
+          name: javah
+          path: target/native
+
+  ubuntu_natives:
+    needs: javah
+    runs-on: ubuntu-latest
+    name: Ubuntu Natives (${{ matrix.arch }})
+    strategy:
+      matrix:
+        arch:
+          - amd64
+          - arm64
+          - ppc64el
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Get JNI headers
+        uses: actions/download-artifact@v4
+        with:
+          name: javah
+          path: target/native
+
+      - name: Build natives in Docker
+        run: resources/ubuntu-build-docker.sh ${{ matrix.arch }} ${{ env.UBUNTU_DIST }} ${{ env.RELEASE_JAVA_VERSION }}
+
+      - name: Upload lib as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: linux-${{ matrix.arch }}
+          path: src/main/resources/*
+
+  mac_natives:
+    needs: javah
+    runs-on: macos-latest
+    name: Mac Natives (${{ matrix.arch }})
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
```

Comment:
Would it make sense to make a "fat" build on macOS? You can build with both architectures into a single dylib, which feels more convenient?

Reply:
With recent Xcode building with multiple `-arch` flags should be enough. If that is problematic somehow, getting the 2 shared libs and combining them with `lipo` is also an option.

Reply:
It's not just about building, but also the loading in the Java part. In which folder do you put that combined dylib?

Reply:
JNA (which we're using for the jitsi-utils JNIUtils) prefers having separate libraries for the different Mac architectures, in separate directories, so I followed their convention.